core/src/main/java/org/apache/iceberg/BaseMetadataTableScan.java

```diff
@@ -19,39 +19,50 @@
 
 package org.apache.iceberg;
 
-import org.apache.iceberg.events.Listeners;
-import org.apache.iceberg.events.ScanEvent;
-import org.apache.iceberg.io.CloseableIterable;
 import org.apache.iceberg.util.PropertyUtil;
-import org.slf4j.Logger;
-import org.slf4j.LoggerFactory;
 
 abstract class BaseMetadataTableScan extends BaseTableScan {
-  private static final Logger LOG = LoggerFactory.getLogger(BaseMetadataTableScan.class);
 
-  protected BaseMetadataTableScan(TableOperations ops, Table table, Schema schema) {
+  private final MetadataTableType tableType;
+
+  protected BaseMetadataTableScan(TableOperations ops, Table table, Schema schema, MetadataTableType tableType) {
     super(ops, table, schema);
+    this.tableType = tableType;
   }
 
-  protected BaseMetadataTableScan(TableOperations ops, Table table, Schema schema, TableScanContext context) {
+  protected BaseMetadataTableScan(TableOperations ops, Table table, Schema schema, MetadataTableType tableType,
+                                  TableScanContext context) {
     super(ops, table, schema, context);
+    this.tableType = tableType;
+  }
+
+  /**
+   * Type of scan being performed, such as {@link MetadataTableType#ALL_DATA_FILES} when scanning
+   * a table's {@link org.apache.iceberg.AllDataFilesTable}.
+   * <p>
+   * Used for logging and error messages.
+   */
+  protected MetadataTableType tableType() {
+    return tableType;
+  }
+
+  @Override
+  public TableScan appendsBetween(long fromSnapshotId, long toSnapshotId) {
```

Comment:
None of the metadata tables support incremental scans so we can share this logic.